Added initial C++ code

Effanuel opened the pull request:
### References:
* Functions used for blockchain: https://github.com/Effanuel/Blockchain/blob/master/Blockchain/Functions.h

### Issue:

### Changes:
* Added functions used for generating users, validating and generating transactions on the blockchain

Commit:
Added code for review

Electrostasy reviewed (commented):
I find many of the comments here could be considered redundant given that the code is fairly self-documenting, but it could use some refactoring/renaming for further improvement to readability, maintainability and correctness.

Additionally, I recommend running this through a code formatter, there's gaps between lines contributing to much wasted space.

Commit:
Added updated file after review

Commit:
Review - renaming

Electrostasy committed:
Fix incorrect use of `using` statement

My bad from reviewing, I forgot C++ has inconsistent `using` syntax, this code should now compile.
